WordPress/Classes/Services/ReaderTopicService.m

```diff
@@ -551,16 +551,7 @@ - (void)siteTopicForSiteWithID:(NSNumber *)siteID
 
         NSManagedObjectID * __block topicObjectID = nil;
         [self.coreDataStack performAndSaveUsingBlock:^(NSManagedObjectContext *context) {
-            ReaderSiteTopic *siteTopic = nil;
-            if (isFeed) {
-                siteTopic = [ReaderSiteTopic lookupWithFeedID:siteID inContext:context];
-            } else {
-                siteTopic = [ReaderSiteTopic lookupWithSiteID:siteID inContext:context];
-            }
-            if (siteTopic) {
-                [context deleteObject:siteTopic];
-            }
-
+            // always upsert new data to existing site topic.
```

Comment:
> Note that removing this may reintroduce the bug in https://github.com/wordpress-mobile/WordPress-iOS/pull/22885, but
> I guess not crashing is a better choice for now.

That's a concern.

In the end, I would argue that this crash is a presentation <-> services layer integration issue, coming from a misuse of Core Data. That's not to say that the services layer doesn't have issues or overcomplications but It will be too easy to reintroduce this crash in the presentation layer by making seemingly unrelated improvements within the service.

Are there any other options that you considered? 🤔 For example, could we make sure that we're always using the newest post within a post view rather than holding a Core Data entity with a possibility of invalidation?

Reply:
After writing that commit, I did a quick test following the steps described in #22885 and it doesn't seem to occur. 

I'll do another pass later and will confirm if this regresses or not. After all, the root cause for #22885 was always returning the cached object, while this commit ensures that the cache is up-to-date.

Reply:
Thanks for letting me know. I'll also retest these regression cases.